File: plugins/eclipse/online.xprei.ide.eclipse/src/online/xprei/ide/eclipse/XpreiSecrets.java
package online.xprei.ide.eclipse;

import org.eclipse.equinox.security.storage.ISecurePreferences;
import org.eclipse.equinox.security.storage.SecurePreferencesFactory;
import org.eclipse.equinox.security.storage.StorageException;

/**
 * API keys in Equinox Secure Storage (OS-backed encryption), never in
 * {@link XpreiSettings} or plaintext — matches VS Code's SecretStorage and
 * the JetBrains plugin's PasswordSafe usage.
 */
public final class XpreiSecrets {
    private static final String NODE = "online.xprei.ide.eclipse/providers";

    private XpreiSecrets() {}

    private static ISecurePreferences node() {
        return SecurePreferencesFactory.getDefault().node(NODE);
    }

    public static String get(String providerId) {
        try {
            return node().get(providerId, null);
        } catch (StorageException e) {
            return null;
        }
    }

    public static void set(String providerId, String apiKey) {
        try {
            ISecurePreferences n = node();
            n.put(providerId, apiKey, true);
            n.flush();
        } catch (Exception e) {
            // Best-effort — not fatal to the current session.
        }
    }

    public static void remove(String providerId) {
        ISecurePreferences n = node();
        n.remove(providerId);
        try {
            n.flush();
        } catch (Exception e) {
            // Best-effort.
        }
    }
}

File: plugins/intellij/src/main/kotlin/online/xprei/ide/XpreiSecrets.kt
package online.xprei.ide

import com.intellij.credentialStore.CredentialAttributes
import com.intellij.credentialStore.Credentials
import com.intellij.credentialStore.generateServiceName
import com.intellij.ide.passwordSafe.PasswordSafe

// API keys in the OS-backed PasswordSafe, never in XpreiSettingsState or
// plaintext — matches VS Code's SecretStorage usage in providers/registry.ts.
object XpreiSecrets {
    private fun attributes(providerId: String): CredentialAttributes =
        CredentialAttributes(generateServiceName("xpreiIDE", providerId))

    fun get(providerId: String): String? =
        PasswordSafe.instance.getPassword(attributes(providerId))

    fun set(providerId: String, apiKey: String) {
        PasswordSafe.instance.set(attributes(providerId), Credentials(providerId, apiKey))
    }

    fun remove(providerId: String) {
        PasswordSafe.instance.set(attributes(providerId), null)
    }
}
